### DISCUSSION TITLE
Unable to resolve type when using styles

### DISCUSSION THREAD

**Discussion by itslyn** [AvaloniaUI/Avalonia#13150] [Q&A] [answered]
Hi. I'm looking at the [Avalonia documentation](https://docs.avaloniaui.net/docs/next/guides/styles-and-resources/how-to-use-included-styles) for using styles. I'm attempting to create my own but have had the same issues, so I have copied the style code from the documentation to see if perhaps I inputted it wrong. However, it's still not working and is saying that it cannot resolve type TextBlock and find a suitable setter property for property Resources.

Here is my code (EDIT: it looks like the code blocks aren't working for me so I'll upload an image as well):
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/e7df8239-3807-47b1-a203-54faa59eadb1)

`<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="MacroMaker.Workspace"
			 Background="Transparent">
	<Grid>
		<Grid.RowDefinitions>
			<RowDefinition Height="*"/>
		</Grid.RowDefinitions>
		<Grid.ColumnDefinitions>
			<ColumnDefinition Width="2*"/>
			<ColumnDefinition Width="*"/>
		</Grid.ColumnDefinitions>
		
		<Rectangle Grid.Row="0"
				   Grid.Column="1"
				   Fill="Black"
				   Opacity="0.5"/>

		<StackPanel Grid.Row="0"
				Grid.Column="1">
			<StackPanel.Resources>
				<Style Selector="TextBlock.h1">
					<Setter Property="FontSize" Value="24"/>
					<Setter Property="FontWeight" Value="Bold"/>
				</Style>
			</StackPanel.Resources>

			<TextBlock Classes="h1">I'm a Heading!</TextBlock>
		</StackPanel>
	</Grid>
</UserControl>
`

However, I am getting the following errors:
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/661ac67b-f8dd-47c4-ba86-a6a12535b8f7)

**Comment by rabbitism** [comment, marked as answer]
Place Style inside StackPanel.Styles
